### ISSUE TITLE
Difficulty moving through certain terrain

### ISSUE THREAD

**Issue by geist22** [swfans/swars#122]
Agents are repeatedly getting stuck in various places when trying to move through certain terrain. It's most easily reproducible in the flower beds (e.g. Syndicate 03: Geneva), but also happens in other places like the flying vehicle platforms in Matochkin Shar. I've recorded a few videos to illustrate the problem more clearly:

https://github.com/swfans/swars/assets/107389189/d30df970-6b43-4dc4-9035-8a822e15f4b7

https://github.com/swfans/swars/assets/107389189/ee9e03c5-c618-4554-b9f5-9b5b3db34c05

In the original, agents have no trouble moving through those places, so this is a bug exclusive to the port.

This seems to be an older bug. The oldest release I've checked so far is [0.3.3.797](https://github.com/swfans/swars/releases/tag/0.3.3.797) and it already has this problem.

**Comment by Moburma** [comment]
I tried with [0.3.2.201](https://github.com/swfans/swars/releases/tag/0.3.1.201) and that does not have the problem, so it's somewhere between that and 797...

**Comment by mefistotelis** [comment]
Hm, I'm confused.. I was sure this is a problem existing in the port from its very start.

**Comment by Moburma** [comment]
The getting stuck on the grass verge problem in the video above 100% definitely doesn't happen for me with with 201, but does on later builds from 797.

**Comment by mefistotelis** [comment]
So we have two problems?

I only checked stopping on texture borders of the landing pad.
But the collisions with other 'things' - including trees and flowers - are a different part of code.

So, very likely two separate issues.

**Comment by geist22** [comment]
> [0.3.2.201](https://github.com/swfans/swars/releases/tag/0.3.1.201)

I've checked this and it looks to me like it works as it's supposed to in both cases. I didn't notice any of the obvious "running against invisible wall" issues there.

**Comment by geist22** [comment]
Phew, that took a while to compile all those commits.

I think this bug was introduced with https://github.com/swfans/swars/commit/7d89a5b93a70bd7a675c5713f2942e0cf079483f

**Comment by geist22** [comment]
I did a bunch of tests. These militia aggro once you move through the front gate past a point around where the walls go off to the side. But if you teleport behind them, they stay inactive unless you move into their weapon range or attack/kill one of them.

In the original game, their behaviour is the same as in the old versions. In the new versions, they also consider the terrain behind them, and within a much larger area (not the full base, but around 2/3 of it). Is it a bug introduced in a new version, or something that was always bugged in the original and is now fixed? I don't know.

I've tracked down the change to https://github.com/swfans/swars/commit/295c4a511b3393f82b2d890f78db26a4cc65a45f

**Comment by mefistotelis** [comment]
Great, that narrows it enough to start analysis.

There are 5 mercenaries on the side with landing pads:

```
Thing(453,1047) group 7 at (17775,0,5338) type=3,6 cmd=0
Thing(454,1048) group 7 at (17760,0,3931) type=3,6 cmd=0
Thing(455,1049) group 7 at (12764,0,8907) type=3,6 cmd=272
Thing(456,1050) group 7 at (12768,0,5818) type=3,6 cmd=266
Thing(457,1051) group 7 at (12791,0,3701) type=3,6 cmd=269
```

Only 3 have commands assigned, and the first commands are all `TNG_CMD_WAIT_MEM_G_ARRIVE` waiting for the player to get within range.

The issue was - starting point for the range was `(1201,-70)` - had a negative coord. This caused the activation circle to be outside of the level.

The updated commands code fixes negative coords, so the range circle works.

In other words, it's a fix.
Thanks for bisecting!

**Comment by geist22** [comment]
As this is fixed, this issue can be closed.
